PublicMapping/districtbuilder
# Import w/ project template

## Overview

Allows selecting a project template during organization import

### Checklist


### Demo

![import-template](https://user-images.githubusercontent.com/4432106/135921571-451f90e3-44be-4092-ace6-86bc5ed1e573.gif)


### Notes

A bit going on here to make this feature work.
The general workflow is:
 - Upload a CSV, it gets validated against the primary state `RegionConfig`
 - We also check to see if the user's organizations have project templates that match the CSV. If so we allow to select an organization/template
 - After a template is selected, we _re_-upload the CSV to validate it against the `RegionConfig` for the template, which may be different (a county of the state in question, or one w/ different data or split census blocks).

## Testing Instructions

- Use `scripts/load-dev-data` to setup your `region_config` table (`TRUNCATE region_config CASCADE;` is a quick way to nuke your current regions first)
- The `scripts/load-dev-data` test organization has two templates configured, one for PA and one for Dane County.
  - Create a map for each template, and export them each to CSV
- Attempt to import each CSV. For the PA template, you should be able to import it w/ the template or using the "No Organization" option. For the Dane County CSV, attempt to import w/o a template will show warnings, which should go away after selecting the template

Closes #971
Closes #1019 

Supersedes #1003
